src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -284,7 +288,15 @@ void CoreWorkerDirectTaskReceiver::HandlePushTask(
           }
         }
       }
-
+      // If we spilled any return objects to plasma, notify the raylet to pin them.
+      // The raylet will then coordinate with the caller to manage the objects'
+      // lifetimes.
+      // TODO(edoakes): the plasma objects could be evicted between creating them
+      // here and when raylet pins them.
+      if (!plasma_return_ids.empty()) {
```

Comment:
Nit, but CoreWorker::ExecuteTask seems like a better place to put this code, since that is where objects are actually put into plasma (maybe after [this block](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/core_worker.cc#L926)).

Reply:
Agree that it's probably better but started doing this and it's a little messy because we have to pipe some things around. Going to leave it as-is for now and we should clean this whole codepath up once the raylet execution path can be removed.